File: Web/src/models/responses/BaseResponse.ts
export enum EResponse {
  Unknown,
  Common,
  Rank,
  Currency,
  Error,
  FromTo,
  LinkZip,
  Radix,
  Temperature,
}

export class BaseResponse {
  public type: EResponse;

  protected constructor(type: EResponse) {
    this.type = type;
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  public Initialize(_: any) {
    throw new Error("Not implemented!");
  }
}

File: Shared/Responses/BaseResponse.cs
namespace Shared.Responses
{
public class BaseResponse
{
    public enum EType
    {
        Unknown,
        Common,
        Rank,
        Currency,
        Error,
        FromTo,
        LinkZip,
        Radix,
        Temperature
    }

    public EType Type { get; }

    protected BaseResponse(EType type)
    {
        Type = type;
    }
}
}
